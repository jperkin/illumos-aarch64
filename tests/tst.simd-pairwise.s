/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.6.8 AdvSIMD scalar pairwise
 */
.macro simd_pairwise op s=0
	.if \s
	\op	s0, v0.2s
	\op	s31, v31.2s
	.endif
	\op	d0, v0.2d
	\op	d31, v31.2d
.endm

.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	simd_pairwise addp
	simd_pairwise fmaxnmp 1
	simd_pairwise faddp 1
	simd_pairwise fmaxp 1
	simd_pairwise fminnmp 1
	simd_pairwise fminp 1
.size libdis_test, [.-libdis_test]
