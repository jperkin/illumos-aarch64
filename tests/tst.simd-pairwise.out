    libdis_test:      00 b8 f1 5e  addp d0, v0.2d
    libdis_test+0x4:  ff bb f1 5e  addp d31, v31.2d
    libdis_test+0x8:  00 c8 30 7e  fmaxnmp s0, v0.2s
    libdis_test+0xc:  ff cb 30 7e  fmaxnmp s31, v31.2s
    libdis_test+0x10: 00 c8 70 7e  fmaxnmp d0, v0.2d
    libdis_test+0x14: ff cb 70 7e  fmaxnmp d31, v31.2d
    libdis_test+0x18: 00 d8 30 7e  faddp s0, v0.2s
    libdis_test+0x1c: ff db 30 7e  faddp s31, v31.2s
    libdis_test+0x20: 00 d8 70 7e  faddp d0, v0.2d
    libdis_test+0x24: ff db 70 7e  faddp d31, v31.2d
    libdis_test+0x28: 00 f8 30 7e  fmaxp s0, v0.2s
    libdis_test+0x2c: ff fb 30 7e  fmaxp s31, v31.2s
    libdis_test+0x30: 00 f8 70 7e  fmaxp d0, v0.2d
    libdis_test+0x34: ff fb 70 7e  fmaxp d31, v31.2d
    libdis_test+0x38: 00 c8 b0 7e  fminnmp s0, v0.2s
    libdis_test+0x3c: ff cb b0 7e  fminnmp s31, v31.2s
    libdis_test+0x40: 00 c8 f0 7e  fminnmp d0, v0.2d
    libdis_test+0x44: ff cb f0 7e  fminnmp d31, v31.2d
    libdis_test+0x48: 00 f8 b0 7e  fminp s0, v0.2s
    libdis_test+0x4c: ff fb b0 7e  fminp s31, v31.2s
    libdis_test+0x50: 00 f8 f0 7e  fminp d0, v0.2d
    libdis_test+0x54: ff fb f0 7e  fminp d31, v31.2d
